openpype/hosts/houdini/api/plugin.py

```diff
@@ -84,3 +90,109 @@ def process(self):
+        """Create node representing instance.
+
+        Arguments:
+            node_name (str): Name of the new node.
+            parent (str): Name of the parent node.
+            node_type (str, optional): Type of the node.
+
+        Returns:
+            hou.Node: Newly created instance node.
+
+        """
+        parent_node = hou.node(parent)
+        instance_node = parent_node.createNode(
+            node_type, node_name=node_name)
+        instance_node.moveToGoodPosition()
+        return instance_node
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        try:
+            if pre_create_data.get("use_selection"):
+                self.selected_nodes = hou.selectedNodes()
+
+            # Get the node type and remove it from the data, not needed
+            node_type = instance_data.pop("node_type", None)
+            if node_type is None:
+                node_type = "geometry"
+
+            instance_node = self._create_instance_node(
+                subset_name, "/out", node_type)
+
+            # wondering if we'll ever need more than one member here
+            # in Houdini
+            instance_data["members"] = [instance_node.path()]
+            instance_data["instance_node"] = instance_node.path()
+
+            instance = CreatedInstance(
+                self.family,
+                subset_name,
+                instance_data,
+                self)
+            self._add_instance_to_context(instance)
+            imprint(instance_node, instance.data_to_store())
+            return instance
+
+        except hou.Error as er:
+            six.reraise(
+                OpenPypeCreatorError,
+                OpenPypeCreatorError("Creator error: {}".format(er)),
+                sys.exc_info()[2])
+
+    def lock_parameters(self, node, parameters):
+        """Lock list of specified parameters on the node.
+
+        Args:
+            node (hou.Node): Houdini node to lock parameters on.
+            parameters (list of str): List of parameter names.
+
+        """
+        for name in parameters:
+            try:
+                parm = node.parm(name)
+                parm.lock(True)
+            except AttributeError:
+                self.log.debug("missing lock pattern {}".format(name))
+
+    def collect_instances(self):
+        for instance in list_instances(creator_id=self.identifier):
+            created_instance = CreatedInstance.from_existing(
+                read(instance), self
+            )
+            self._add_instance_to_context(created_instance)
+
+    def update_instances(self, update_list):
+        for created_inst, _changes in update_list:
+            instance_node = hou.node(created_inst.get("instance_node"))
+            current_data = read(instance_node)
+
+            imprint(
+                instance_node,
+                {
+                    key: value[1] for key, value in _changes.items()
+                    if current_data.get(key) != value[1]
```

Comment:
```suggestion
                    if current_data.get(key) != value[0]
```

Reply:
I was implementing at Nuke and found out index was working this way rather then to `1`

Reply:
@jakubjezek001 That should be wrong. This should actually read:

```python
                {
                    key: new_value for key, (old_value, new_value) in _changes.items()
                    if current_data.get(key) != new_value
                }
```

But might be more readable by not doing a dict comprehension, something like:
```python
updates = {}
for key, (old_value, new_value) in _changes:
    if key not in current_data or new_value != current_data[key]:
        updates[key] = new_value
        
imprint(instance_nodes, updates, update=True)
```